pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/table/internal/JavaResolvers.java

```diff
@@ -73,6 +74,39 @@ static boolean canBeImportedIn(String thisPackage, JAccessibleElementSymbol memb
         return isAccessibleIn(null, thisPackage, member, false);
     }
 
+    @NonNull
+    static NameResolver<JTypeMirror> moduleImport(Set<String> moduleNames,
+                                                  final SymbolResolver symbolResolver,
+                                                  final String thisPackage) {
+        return new SingleNameResolver<JTypeMirror>() {
+            @Override
+            public @Nullable JTypeMirror resolveFirst(String simpleName) {
+                for (String module : moduleNames) {
+                    JModuleSymbol moduleSymbol = symbolResolver.resolveModule(module);
```

Comment:
This should be cached in the name resolver, otherwise it will perform a classloader lookup and parse the module file for every name that is requested (which is probably all the type identifiers in the compilation unit).